web-ui/app/js/mail_view/ui/mail_view.js

```diff
@@ -96,72 +96,79 @@ define(
       };
 
       this.checkEncrypted = function(mail) {
-        if(_.isEmpty(mail.security_casing.locks)) {
-          return {
-            cssClass: 'security-status__label--not-encrypted',
-            label: 'not-encrypted'
-          };
-        }
-
-        var statusClass = ['security-status__label--encrypted'];
+        var statusClass = ['security-status__label'];
         var statusLabel;
+        var tooltip;
 
-        var hasAnyEncryptionInfo = _.any(mail.security_casing.locks, function (lock) {
-          return lock.state === 'valid';
-        });
-
-        if(hasAnyEncryptionInfo) {
-          statusLabel = 'encrypted';
+        if(_.isEmpty(mail.security_casing.locks)) {
+          statusClass.push('--not-encrypted');
+          statusLabel = 'not-encrypted';
+          tooltip = 'This message is not encrypted.';
         } else {
-          statusClass.push('--with-error');
-          statusLabel = 'encryption-error';
+          statusClass.push('--encrypted');
+
+          var hasAnyEncryptionInfo = _.any(mail.security_casing.locks, function (lock) {
+            return lock.state === 'valid';
+          });
+
+          if(hasAnyEncryptionInfo) {
+            statusLabel = 'encrypted';
+            tooltip = 'This message was encrypted.';
+          } else {
+            statusClass.push('--with-error');
+            statusLabel = 'encryption-error';
+            tooltip = 'This message had an encryption error.';
```

Comment:
Okay, so I have two suggestions for how to refactor this. We have three smells here, it seems to me:

* Repetition of code for statusLabel, statusClass and tooltip
* Unnecessary mutable code
* Nested ifs

I'm going to try to make refactor suggestions, more for the readability and learning potential then anything else. Let me know if this makes sense to you.

### Repetition of vars
The end goal is to return a structure of vars that will change completely depending on the flag we want. So, why not have three different structures:

```
ENCRYPTED_FLAG = {
  statusClass: 'security-status__label',
  statusLabel: 'encrypted',
  tooltip: 'This message was encrypted.'
}
ENCRYPTED_WITH_ERROR_FLAG = ...
UNENCRYPTED_FLAG = ...
```

### Unnecessary mutable code
The solution above would also solve this problem, so that now we won't be changing the vars or pushing to them, but actually returning static structures.

### Nested ifs
With the suggestions above, now we have code looking something like:
```
if(_.isEmpty(mail.security_casing.locks)) { return UNENCRYPTED_FLAG; }
if(hasAnyEncryptionInfo) { return ENCRYPTED_FLAG; }
return ENCRYPTED_WITH_ERROR_FLAG;
```

Much more readable, no? Thoughts?

Same thing works for the signature flag.